Dockerfile

```
@@ -0,0 +1,28 @@
+# Prepare the boot build image and dependencies
+FROM clojure:boot-alpine AS boot
+RUN apk add --update git npm
+RUN npm install -g yarn shadow-cljs
+
+# Copy source files into the docker container
+WORKDIR /app
+COPY . ./
+
+# Update game data to the latest release
+RUN git submodule update --init
+RUN cd resources/game \
+ && git checkout master \
+ && git checkout $(git describe --tags --abbrev=0 $(git rev-list --tags --max-count=2) | grep -vE 'continuous')
```

Seems like `max-count` should be 1 in order for `git checkout` to work.

```suggestion
 && git checkout $(git describe --tags --abbrev=0 $(git rev-list --tags --max-count=1) | grep -vE 'continuous')
```

---

Also I'm not really sure about the idea of always grabbing the latest version. My development cycle for this project assumes that I update the submodule with the game files to the latest stable version, then try to build the website, fixing all the errors that arise in the process, then commit all my changes (including the updated reference to the game repository) to git. This way code is guaranteed to work with the game files since version of the code always corresponds to version of the game files.

I would prefer to switch game versions manually before building the docker image.

---

`max-count=2` is necessary because `continuous` is always in the top 2 results and needs to be filtered out. I was hoping to find a more... exact way to locate the latest release version but this was the best I found.

On switching game versions, I think the manual switch flow makes more sense for doing local builds. But my idea for Docker as a self-contained instance was more targeted at supporting the use case of "I just want to get this running without worrying about setup", which would be more likely to want the current version.

---

On my machine top 2 results (output of `git describe --tags --abbrev=0 $(git rev-list --tags --max-count=2)`) are `v0.9.16.1` and `v0.9.16` while the `continuous` tag points to an old commit https://github.com/endless-sky/endless-sky/commit/c73331d014eaa7bf4ca9bceb429c6ce49e52e0da. Seems like the endless-sky team is moving the `continuous` tag from commit to commit (which is considered a really bad practice since tags are supposed to be immutable), so this is unreliable.

If you want the latest release version, you can use `git checkout $(git tag -l --sort=-v:refname | head -n 1)`.

---

Oops, this command may decide that `continuous` is the latest tag. The correct command would be `git checkout $(git tag -l --sort=-v:refname "v*" | head -n 1)`

---

Ooh, that's great, thanks!